[CADMIN] Creating CADMIN_1_25 python test module

j-ororke opened the pull request:
Created TC_CADMIN_1_25:
- Creating TC_CADMIN_1_25 python3 test module for task [#287](https://github.com/project-chip/matter-test-scripts/issues/287) with test step in test plan PR [#4986](https://github.com/CHIP-Specifications/chip-test-plans/pull/4986)
- Removing Test_TC_CADMIN_1_25.yaml script

#### Testing
WSL
TH UI
REPL

Raspberry Pi REPL:
- chip-cert-bins: f99a2c3b16d3d26f91a0c5efb2770d1c488697d1

TH UI:
- version: spring2025 
- SHA: 953169b

j-ororke committed:
[CADMIN] Creating CADMIN_1_25 python test module:
- Creating TC_CADMIN_1_25 python3 test module for task #287

j-ororke committed:
Updating TC_CADMIN_1_25 python3 test module:
- Changed verbage to correct verification result for test step 31
- Updated to using revoke commissioning command in place of waiting for commissioning timeouts to occur as those are already tested in other CADMIN automated tests.

restyled-commits committed:
Restyled by autopep8

restyled-commits committed:
Restyled by isort

j-ororke committed:
Updating TC_CADMIN_1_25 python3 test module:
- Resolved linting errors

j-ororke committed:
Updating TC_CADMIN_1_25.py:
- Resolving linting errors

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Removing Test_TC_CADMIN_1_25.yaml:
- Removed yaml test script as python3 test module has been created.

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_25_test

cecille commented on src/python_testing/TC_CADMIN_1_25.py:
```diff
@@ -0,0 +1,606 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
```
```suggestion
#    Copyright (c) 2025 Project CHIP Authors
```

cecille reviewed (commented):
Note - not done with this review yet, but I think the switch over to use the already available classes is going to cause enough of a change here that it's worthwhile to just start there.

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_25_test

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_25_test

j-ororke committed:
Updating TC_CADMIN_1_25 python3 test module:
- Now using accumulatorClusterAttributeChangeAccumulator to establish subscriptions to attributes
- Updated to reset accumulator queues inbetween each test validation for verifying new values are found
- Using await_all_expected_report_matches() to verify that new values match what we are looking for
- Using AttributeMatcher.from_callable() to verify that we have the correct values to match against our newly found values

j-ororke commented on src/python_testing/TC_CADMIN_1_25.py:
```diff
@@ -0,0 +1,606 @@
+        fabrics = await self.read_single_attribute_check_success(dev_ctrl=th, fabric_filtered=False, endpoint=0, cluster=OC_cluster, attribute=OC_cluster.Attributes.Fabrics)
+        return fabrics
+
+    async def wait_for_subscription_value(self, queue_obj, expected_value=None, timeout=10):
```
Hi Cecille, 
My apologies, I have removed the prior locally created subscription method in the test module and updated to using the subscription handler classes and accumulators in the matter_testing support module for validation of this test.

cecille replied:
thanks!

j-ororke committed:
Apply autopep8 and isort formatting

j-ororke committed:
Resolving linting errors

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_25_test

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_25_test

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_25_test
